### PR TITLE
Create general format for the shell prooject

### PR DESCRIPTION
**Pull request by rakati**
- Set a Minimal structure for the Shell project as discussed in our meeting,
- Add Authors file.

*N.B. The structure is still minimal and can be changed later.*

### CODE REVIEW

File: shell.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+#include <unistd.h>
+#include <fcntl.h>
+
+/*
+ * String Utility functions
+ */
+
+int _strlen(char *);
+void _puts(char *);
+int _getline(const int fd, char **line);
```
Comment (amine-zzr) [inline shell.h:13]: is it okay to reverse the function parameters or is this a mistyping ?

File: shell.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ * @SEM: indicating relation with next command is ';' operator.
+ * @END: means no command after current command.
+ */
+enum rel
+{
+	OR,
+	AND,
+	SEM,
+	END
+};
```
Comment (amine-zzr) [inline shell.h:32]: i didn't know about this data type yet, i will try to learn about it
but is this data type allowed to use, i didn't read the entire project

File: shell.c [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ * main - Entry point for our shell program
+ *
+ * @ac: (integer) number of argument of the program
+ * @av: (array of strings) array of parameters of the program with len=ac
+ * @envp: (array of strings) array of environment variable path.
+ * @return int 0 for success otherwise error number
+ */
+int main(int ac, char **av, char **envp)
+{
+	char line;
```
Comment (amine-zzr) [inline shell.c]: is `line` a pointer or a character?

File: shell.h (hunk 1 above)
Comment (rakati) [reply shell.h:13]: yes it is okay, it is our getline

File: shell.h (hunk 2 above)
Comment (rakati) [reply shell.h:32]: it is allowed, enums are useful, and it is good to learn about them

File: shell.c (hunk 3 above)
Comment (rakati) [reply shell.c]: good catch, it should be a pointer.

File: shell.h (hunk 1 above)
Comment (amine-zzr) [reply shell.h:13]: that's not what i meant.. in the main function you have passed the address of the pointer as the first argument, then the fd variable as the second argument, but in the prototype you have passed the fd as the first argument then the address of the line pointer as the second argument

Comment (rakati) [reply shell.h:13]: fixed by [b71c539](https://github.com/rakati/simple_shell/pull/1/commits/b71c539817de2eec41d4553edfd570355c4833d2)

File: shell.c (hunk 3 above)
Comment (rakati) [reply shell.c]: fixed by [e9b088c](https://github.com/rakati/simple_shell/pull/1/commits/e9b088c1b8813a23ff37e30dac7754ad10274276)

**Pull request merged**